src/ast/passes/semantic_analyser.cpp

```diff
@@ -126,6 +126,10 @@ void SemanticAnalyser::visit(Identifier &identifier)
     identifier.type = CreateInt(
         std::get<0>(getIntcasts().at(identifier.ident)));
   }
+  else if (func_ == "offsetof")
+  {
+    identifier.type = CreateString(identifier.ident.size() + 1);
+  }
```

Comment:
I'm not sure what the purpose of this code is. Why not set the args types when analysing the call?